pkg/project/project.go

```diff
@@ -76,28 +76,29 @@ func WithDetection(c Config) heartbeat.HandleOption {
 					continue
 				}
 
-				result := Detect(h.Entity, c.MapPatterns)
+				result := Detect(h.Entity, config.MapPatterns)
 
 				if result.Project == "" {
 					result.Project = h.ProjectOverride
 				}
 
 				if result.Project == "" || result.Branch == "" {
-					revControlResult := DetectWithRevControl(h.Entity, c.SubmodulePatterns, c.ShouldObfuscateProject)
+					revControlResult := DetectWithRevControl(h.Entity, config.SubmodulePatterns)
 
 					result.Branch = firstNonEmptyString(result.Branch, revControlResult.Branch)
 					result.Folder = firstNonEmptyString(result.Folder, revControlResult.Folder)
-					result.Project = firstNonEmptyString(result.Project, revControlResult.Project)
 
-					if result.Project == "" {
-						result.Project = setProjectName(h.ProjectAlternate, c.ShouldObfuscateProject, revControlResult.Folder)
+					if config.ShouldObfuscateProject && revControlResult.Project != "" && result.Project == "" {
```

Comment:
I guess here it would be helpful to explain in a comment, in which cases we obfuscate and in which not. Just to provide some more info on the underlying logic.

Reply:
yep sounds good.